test/Libraries/DynamoPythonTests/PythonEditTests.cs

```diff
@@ -511,5 +512,107 @@ public void Python_CanReferenceDynamoServicesExecutionSession()
                 AssertPreviewCount(guid, 2);
             }
         }
+
+        [Test]
+        public void CPythonClassCanBeUsedInDownStreamNode()
+        {
+            // open test graph
+            var examplePath = Path.Combine(TestDirectory, @"core\python", "cpythoncustomclass.dyn");
+            ViewModel.OpenCommand.Execute(examplePath);
+
+            var downstream1 = ViewModel.Model.CurrentWorkspace.Nodes.First(x => x.Name == "downstream1");
+            var downstream2 = ViewModel.Model.CurrentWorkspace.Nodes.First(x => x.Name == "downstream2");
+            
+            ViewModel.HomeSpace.Run();
+            AssertPreviewValue(downstream1.GUID.ToString(), "firstName");
+            AssertPreviewValue(downstream2.GUID.ToString(), "firstNamelastname");
+
+        }
+        [Test]
+        public void CPythonClassCanBeModifiedInDownStreamNode()
+        {
+            // open test graph
+            var examplePath = Path.Combine(TestDirectory, @"core\python", "cpythoncustomclass_modified.dyn");
+            ViewModel.OpenCommand.Execute(examplePath);
+
+            var downstream1 = ViewModel.Model.CurrentWorkspace.Nodes.First(x => x.Name == "downstream1");
+            var downstream2 = ViewModel.Model.CurrentWorkspace.Nodes.First(x => x.Name == "downstream2");
+
+            ViewModel.HomeSpace.Run();
+            AssertPreviewValue(downstream2.GUID.ToString(), "joe");
+        }
+
+        [Test]
+        public void TwoCPythonHandlesReturnedFromSameNodeHaveSameHandleID()
+        {
+            // open test graph
+            var examplePath = Path.Combine(TestDirectory, @"core\python", "cpythoncustomclass_returnManyInstances.dyn");
+            ViewModel.OpenCommand.Execute(examplePath);
+
+            var classdef = ViewModel.Model.CurrentWorkspace.Nodes.First(x => x.Name == "classdef");
+
+            ViewModel.HomeSpace.Run();
```

Comment:
Isn't the graph run in line 550 itself?

Reply:
yes this is probably overkill, but I want to guard against this graph being made manual at some point by another dev.

Reply:
if it's confusing I can remove it.

Reply:
or make the graph manual, so the runs are more explicit.

Reply:
I guess there's no harm as if there is no change in the graph, it will effectively not run.